README.md

```diff
@@ -13,6 +13,164 @@
 This repo is to support
 https://github.com/scikit-build/scikit-build-core/issues/230.
 
+## For users
+
+Every external plugin must specify a "provider", which is a module that provides
+the API listed in the next section.
+
+```toml
+[tool.dynamic-metadata.v1]
+<field-name>.provider = "<module>"
+```
+
+There is an optional field: "provider-path", which specifies a local path to
+load a plugin from, allowing plugins to reside inside your own project.
+
+All other fields are passed on to the plugin, allowing plugins to specify custom
+configuration per field. Plugins can, if desired, use their own `tool.*`
+sections as well; plugins only supporting one metadata field are more likely to
+do this.
+
+### Example: regex
+
+An example regex plugin is provided in this package. It is used like this:
+
+```toml
+[build-system]
+requires = ["...", "dynamic-metadata"]
+build-backend = "..."
+
+[project]
+dynamic = ["version"]
+
+[tool.dynamic-metadata.v1.version]
```

Comment:
... because it may be confusing to see `.v1.version` without providing some context.